lib/libc/gen/tls_malloc.c

```diff
@@ -204,8 +204,10 @@ __rederive_pointer(void *ptr)
 	TLS_MALLOC_LOCK;
 	for (i = 0; i < n_pagepools; i++) {
 		char *pool = pagepool_list[i];
-		if (cheri_is_address_inbounds(pool, addr))
+		if (cheri_is_address_inbounds(pool, addr)) {
```

Comment:
hmm we have three very similar copies of the same file (rtld_malloc.c, tls_malloc.c and libsimple_malloc/malloc.c) and it seems like some changes/fixes were not applied to all of them.

Reply:
Well, rtld_malloc is not used for purecap so that one doesn't really matter.

Reply:
rtld_malloc doesn't have any locking either as I think it's already single-threaded via the bind lock, etc.

libmalloc_simple doesn't  have any locking either.